crates/muvm/src/guest/mount.rs

```diff
@@ -60,33 +91,126 @@ fn mount_fex_rootfs() -> Result<()> {
+                }
+            }
+            assert!(!layers.is_empty());
+            if layers.len() == 1 {
+                do_mount_recursive_bind(&layers[0], target)?;
+            } else {
+                if layers[0] == "/etc" {
+                    // Special case: /etc has an overlaid mount for /etc/resolv.conf,
+                    // which will confuse overlayfs. So grab the raw mount.
+                    layers[0] = "/run/muvm-host/etc".to_owned();
+                }
+                let opts = format!(
+                    "lowerdir={},metacopy=off,redirect_dir=nofollow,userxattr",
+                    layers.into_iter().rev().collect::<Vec<String>>().join(":")
+                );
+                let opts = CString::new(opts).unwrap();
+                let overlay = "overlay".to_string();
+                let overlay_ = Some(&overlay);
+
+                mount2(overlay_, &target, overlay_, flags, Some(&opts))
+                    .context("Failed to overlay")?;
+            }
+        }
+
+        // Special case: Put back the /etc/resolv.conf overlay on top
+        overlay_file(
+            "/etc/resolv.conf",
+            &(dir_rootfs.clone() + "/etc/resolv.conf"),
+        )?;
+    } else {
+        if images.len() >= 2 {
+            // Overlay the mounts together.
+            let opts = format!(
+                "lowerdir={}",
+                images.into_iter().rev().collect::<Vec<String>>().join(":")
+            );
+            let opts = CString::new(opts).unwrap();
+            let overlay = "overlay".to_string();
+            let overlay_ = Some(&overlay);
+
+            mkdir_fex(&dir_rootfs);
+            mount2(overlay_, &dir_rootfs, overlay_, flags, Some(&opts))
+                .context("Failed to overlay")?;
+        } else if images.len() == 1 {
+            assert!(!merged_rootfs);
+            // Just expose the one mount
+            symlink(&images[0], &dir_rootfs)?;
+        } else if images.is_empty() {
+            assert!(!merged_rootfs);
+            // If no images were passed, FEX is either managed by the host os
+            // or is not installed at all. Avoid clobbering the config in that case.
+            return Ok(());
+        }
     }
 
     // Now we need to tell FEX about this. One of the FEX share directories has an unmounted rootfs
     // and a Config.json telling FEX to use FUSE. Neither should be visible to the guest. Instead,
     // we want to replace the folders and tell FEX to use our mounted rootfs
     for base in ["/usr/share/fex-emu", "/usr/local/share/fex-emu"] {
         if Path::new(base).exists() {
-            let json = format!("{{\"Config\":{{\"RootFS\":\"{dir_rootfs}\"}}}}\n");
+            let json = if merged_rootfs {
+                format!("{{\"Config\":{{\"RootFS\":\"{dir_rootfs}\",\"MergedRootFS\":\"1\"}}}}\n")
+            } else {
+                format!("{{\"Config\":{{\"RootFS\":\"{dir_rootfs}\"}}}}\n")
+            };
```

Comment:
```suggestion
            let json = if merged_rootfs {
                json!({
                    "Config": {
                        "RootFS": dir_rootfs,
                        "MergedRootFS": "1",
                    },
                })
            } else {
                json!({
                    "Config": {
                        "RootFS": dir_rootfs,
                    },
                })
            }.to_string();
```

https://docs.rs/serde_json/latest/serde_json/index.html#constructing-json-values
https://docs.rs/serde_json/latest/serde_json/macro.json.html
https://docs.rs/serde_json/latest/serde_json/enum.Value.html#impl-Display-for-Value